cmd/kubeadm/app/phases/selfhosting/selfhosting.go

```diff
@@ -104,9 +104,11 @@ func CreateSelfHostedControlPlane(manifestsDir, kubeConfigDir string, cfg *kubea
 			return err
 		}
 
-		// Remove the old Static Pod manifest
-		if err := os.RemoveAll(manifestPath); err != nil {
-			return fmt.Errorf("unable to delete static pod manifest for %s [%v]", componentName, err)
+		// Remove the old Static Pod manifest if not dryrunning
+		if !dryRun {
+			if err := os.RemoveAll(manifestPath); err != nil {
```

Comment:
We still want static manifests for etcd.

Reply:
yes, but this only touches the core control plane components.
This change is just to not touch files when dryrunning